### PR TITLE
Delete message

### PR DESCRIPTION
**Pull request by gabriel**
This creates a delete message action, and in the chat reducer, on prepend and append, we handle delete message types by keeping list of all deletedIDs, and filtering out messages that are deleted.

### CODE REVIEW

**Comment by gabriel** [comment]
@keybase/react-hackers

File: shared/actions/chat.js [hunk 1]
```diff
@@ -357,6 +358,34 @@ function * _postMessage (action: PostMessage): SagaGenerator<any, any> {
   }
 }
 
+function * _deleteMessage (action: DeleteMessage): SagaGenerator<any, any> {
+  const {message} = action.payload
+  let messageID: ?MessageID
+  let conversationIDKey: ConversationIDKey
+  switch (message.type) {
+    case 'Text':
```
Comment (chrisnojima) [inline shared/actions/chat.js]: We should handle delete of other types also, attachments etc

Comment (chrisnojima) [inline shared/actions/chat.js]: or at least a comment // TODO that we should

File: shared/reducers/chat.js [hunk 2]
```diff
@@ -10,10 +10,10 @@ const {StateRecord, ConversationStateRecord, makeSnippet, serverMessageToMessage
 const initialState: State = new StateRecord()
 const initialConversation: ConversationState = new ConversationStateRecord()
 
-function _dedupeMessages (seenMessages: Set<any>, messages: List<ServerMessage> = List(), prepend: List<ServerMessage> = List(), append: List<ServerMessage> = List()): {nextSeenMessages: Set<any>, nextMessages: List<ServerMessage>} {
+function _dedupeMessages (seenMessages: Set<any>, messages: List<ServerMessage> = List(), prepend: List<ServerMessage> = List(), append: List<ServerMessage> = List(), deletedIDs: Set<any>): {nextSeenMessages: Set<any>, nextMessages: List<ServerMessage>} {
```
Comment (chrisnojima) [inline shared/reducers/chat.js]: maybe this should be renamed. it no long only dedupes

File: shared/actions/chat.js (hunk 1 above)
Comment (gabriel) [inline shared/actions/chat.js]: Added case for 'Attachment'

File: shared/reducers/chat.js (hunk 2 above)
Comment (gabriel) [inline shared/reducers/chat.js]: Renamed to _filterMessages and added comments

File: shared/actions/chat.js [hunk 3]
```diff
@@ -368,6 +368,9 @@ function * _deleteMessage (action: DeleteMessage): SagaGenerator<any, any> {
     case 'Text':
       conversationIDKey = message.conversationIDKey
       messageID = message.messageID
```
Comment (chrisnojima) [inline shared/actions/chat.js]: either need a break or use a fallthrough (since its the same code)

Comment (gabriel) [inline shared/actions/chat.js]: oops, the linter caught this too

**Pull request merged**